### PR TITLE
Add all system cursors (engine)

### PR DESCRIPTION
**Pull request by dkwingsmt**
## Description

This PR adds support to a number of system cursors to the following shells:
- Android
- CSS
- Windows (Win32)
- Linux (GTK)
- macOS

All cursors that are added in https://github.com/flutter/flutter/pull/60931 and https://github.com/flutter/flutter/pull/61399 are included. Some of them might use a fallback.

## Related Issues

- Works in coordination with https://github.com/flutter/flutter/pull/60931, but does not require it to run. 

## Tests

I added the following tests:

- N/A. The changes are only hard-coded values, which should be "tested" by not changing them. Testing all these values will basically be writing all of them a second time.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/windows/win32_flutter_window.cc [hunk 1]
```diff
@@ -18,15 +18,37 @@ constexpr int kScrollOffsetMultiplier = 20;
...
 static HCURSOR GetCursorByName(const std::string& cursor_name) {
   static auto* cursors = new std::map<std::string, const wchar_t*>{
-      {"none", nullptr},
-      {"basic", IDC_ARROW},
-      {"click", IDC_HAND},
-      {"text", IDC_IBEAM},
-      {"forbidden", IDC_NO},
-      {"horizontalDoubleArrow", IDC_SIZEWE},
-      {"verticalDoubleArrow", IDC_SIZENS},
+      {'allScroll' : IDC_SIZEALL},
```
Comment (gspencergoog) [inline shell/platform/windows/win32_flutter_window.cc:26]: I think these all need to use double quotes, not single quotes.

**Comment by SahajRana** [comment]
@dkwingsmt, Does the windows cursor support work now?

**Comment by dkwingsmt** [comment]
Yep! Give a try!

**Comment by SahajRana** [comment]
@dkwingsmt, Yes, it works! Thanks, flutter team for implementing it. 
If we could have the ability to add custom cursors( .cur files) as an asset as well that would be really great.

**Pull request merged**